firmware/hw_layer/board_overrides.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * @file board_overrides.h
+ * @brief Board-specific override mechanism
+ *
+ * This header provides a centralized way to define and manage board-specific overrides.
+ * Custom boards can implement their own specialized behavior by overriding these function pointers.
+ *
+ * Usage Example:
+ *
+ * 1. Define your custom function:
+ *    void myBoardCustomHello() {
+ *        // Your custom implementation
+ *    }
+ *
+ * 2. Set up the override in your board_configuration.cpp:
+ *    void setup_custom_board_overrides() {
+ *        custom_board_boardSayHello = myBoardCustomHello;
+ *    }
+ *
+ *
+ * @date: jul 09, 2025
+ * @author FDSoftware
+ */
+
+#pragma once
+#include <functional>
+#include <optional>
+
+using setup_custom_board_overrides_type = void (*)();
```

Comment:
this cursed `void (*)()` thing has to be read as "function with no parameters and returning void"

Reply:
added comment